src/theme-context-provider.tsx

```diff
@@ -180,28 +190,49 @@ const ThemeContextProvider: React.FC<Props> = ({theme, children, as, withoutStyl
         [colors]
     );
 
-    // TODO: create CSS vars for size and lineHeight (https://jira.tid.es/browse/WEB-1929)
     const textPresetsVars = React.useMemo(() => {
         // Get an object mapping textPresets tokens to objects containing the token's weight
         // For example, {title1: {weight: '700'}}
         const tokenValues = Object.entries(contextTheme.textPresets).map(([token, config]) => {
             // Map light/regular/medium/bold to valid css fontWeight values
-            return {[token]: {weight: String(mapToWeight[config.weight])}};
+            return {
+                [token]: {
+                    weight: String(mapToWeight[config.weight]),
+                    ...(config.size && {size: `${config.size.desktop}px`}),
+                    ...(config.lineHeight && {lineHeight: `${config.lineHeight.desktop}px`}),
+                },
+            };
         });
 
-        const textPresetsVars = Object.assign({}, ...tokenValues) as {
-            [key in keyof TextPresetsConfig]: {weight: string};
-        };
+        return Object.assign({}, ...tokenValues) as TextPresetsVars;
+    }, [contextTheme]);
 
-        return textPresetsVars;
+    const textPresetsResponsiveVars = React.useMemo(() => {
+        const tokenValues = Object.entries(contextTheme.textPresets).map(([token, config]) => {
+            return {
+                [token]: {
+                    weight: String(mapToWeight[config.weight]),
+                    // Use mobile values for size/lineHeight
+                    ...(config.size && {size: `${config.size.mobile}px`}),
+                    ...(config.lineHeight && {lineHeight: `${config.lineHeight.mobile}px`}),
+                },
+            };
+        });
+
+        return Object.assign({}, ...tokenValues) as TextPresetsVars;
     }, [contextTheme]);
 
-    const themeVars = assignInlineVars(vars, {
+    const themeVarsValues = {
         textPresets: textPresetsVars,
         colors,
         rawColors,
         borderRadii: theme.skin.borderRadii ?? defaultBorderRadiiConfig,
-    });
+    };
+
+    const responsiveThemeVarsValues = {
+        ...themeVarsValues,
```

Comment:
does this mean you are repeating all the css vars, even those that aren't responsive (colors, radii, etc)? That would duplicate the size of the inline style

Reply:
Yep, good catch. I've fixed that